EpicLoot/src/Magic/MagicItemEffects/AddEitrLeech.cs

```diff
@@ -0,0 +1,61 @@
+﻿using System;
+using EpicLoot.Healing;
+using HarmonyLib;
+using UnityEngine;
+
+namespace EpicLoot.MagicItemEffects
+{
+    public static class AddEitrLeech
+    {
+        [HarmonyPatch(typeof(Character), nameof(Character.Damage))]
+        public static class AddEitrLeechAddLifeSteal_Character_Damage_Patch
+        {
+            public static void Postfix(HitData hit)
+            {
+                CheckAndDoEitrLeech(hit);
+            }
+        }
+        
+        public static void CheckAndDoEitrLeech(HitData hit)
+        {
+            try
+            {
+                if (!hit.HaveAttacker())
+                {
+                    return;
+                }
+
+                var attacker = hit.GetAttacker() as Humanoid;
+                if (attacker == null)
+                {
+                    return;
+                }
+
+                var weapon = attacker.GetCurrentWeapon();
+                if (Attack_Patch.ActiveAttack != null)
+                    weapon = Attack_Patch.ActiveAttack.m_weapon;
+
+                // in case weapon's durability is destroyed after hit?
+                // OR in case damage is delayed and player hides weapon
+                if (weapon == null || !weapon.IsMagic() || !(attacker is Player player))
+                    return;
```

Comment:
I want to move away from this code style to make it consistent. Can you enclose all single line return statements in brackets:
```csharp
{
    return;
}
```